json_yaml_files/prepare_files.py

```diff
@@ -0,0 +1,178 @@
+from bids import BIDSLayout
+from collections import defaultdict
+import re
+import json
+import sys
+import yaml
+from pathlib import Path
+
+
+class BIDStoNDAConfigGenerator:
+    def __init__(self):
+        self.path = sys.argv[1]
```

Comment:
I know the utility of sys.argv, but it can be dangerous and problematic if an argument is not provided or if more than one argument is provided on accident. Could you write an [argparse.ArgumentParser](https://docs.python.org/3/library/argparse.html) instead? It doesn't have to block this PR, but it should be added as an Issue if this PR is merged before adding and [argparse.ArgumentParser](https://docs.python.org/3/library/argparse.html). Thanks!

Reply:
Sure. I will make the changes in the approach to fetch the value from arguments.